### PR TITLE
Changes to support ARKit exposing the camera frames

### PR DESCRIPTION
**Pull request by judax**
and rendering them in ARView.

##### Changes Proposed:

ARView now is able to determine if iOS is exposing the camera frames and render them similarly as how WebARonARCore does (but using the correct sampler and texture targets, not the external image OES extension)

##### Checklist:

- [X] Tested on WebARonARCore
- [X] Tested on WebARonARKit
- [X] I've read CONTRIBUTING.md and signed the CLA (required)
- [X] This PR does not contain built changes in `dist/\*` (required)
- [X] This PR only contains one commit (required)

### CODE REVIEW

**Review by lincolnfrog — APPROVED** [review]
This looks great, Iker. Excellent work!

**Comment by judax** [comment]
Pushed some changes to adapt to the WebARonARKit review.

File: src/shaders/arview.frag [hunk 1]
```diff
@@ -11,13 +11,11 @@
...
 // limitations under the License.
 
-#extension GL_OES_EGL_image_external : require
-
 precision mediump float;
 
 varying vec2 vTextureCoord;
 
-uniform samplerExternalOES uSampler;
+uniform sampler2D uSampler;
```
Comment (jsantell) [inline src/shaders/arview.frag:18]: Will this break on older installations of WebARonARCore? Wasn't the samplerExternalOES necessary?

File: src/ARView.js [hunk 2]
```diff
@@ -135,7 +137,18 @@ class ARVideoRenderer {
     this.gl = gl;
     if (this.vrDisplay) {
       this.passThroughCamera = vrDisplay.getPassThroughCamera();
-      this.program = getProgram(gl, vertexSource, fragmentSource);
+      // Depending on the type of passThroughCamera, use a different
+      // target for the texture and different shaders.
+      if (this.passThroughCamera instanceof Image) {
```
Comment (jsantell) [inline src/ARView.js:141]: Maybe a comment indicating what's for WebARonARCore and what's for WebARonARKit+WebARonARKitSendsCameraFrames

File: src/ARView.js [hunk 3]
```diff
@@ -248,6 +263,11 @@ class ARVideoRenderer {
     ];
 
     preserveGLState(gl, bindings, () => {
+      // If the camera pass through is still not valid, skip the rendering.
+      if (this.passThroughCamera.textureWidth == 0 ||
```
Comment (jsantell) [inline src/ARView.js]: nit `===`

**Review by jsantell — COMMENTED** [review]
Some comments, excited to see the video texture processing on WebARonARKit! However I'm a bit out of touch with those recent developments, so take my comments with a grain of salt

**Comment by jsantell** [comment]
Nit: linting failure in ARView:   `267:54  error  Trailing spaces not allowed  no-trailing-spaces`

File: src/shaders/arview.frag (hunk 1 above)
Comment (judax) [reply src/shaders/arview.frag:18]: There are still shaders that use the External Image OES extension. WebARonARCore still uses this sampler type, just that the code is in a different shader file (with the -oes suffix).

**Comment by judax** [comment]
Pushed a new revision addressing some of the concerns by @jsantell (that should be enjoying his PTO!!!)

**Pull request merged**